Login design

tianhaoyao opened the pull request:
Implemented login and register with styling as per mockup. Linked to issue #1

WhoaNellie reviewed (commented):
Couple of notes!
- With UI Changes, it's helpful to add screenshots of what the UI looks like to your PR description so reviews can see the style without having to run the code!

- There's a good amount of repeated code/styles between the Login and Signup pages, you should consider what you can make into reusable components

- There's a [theme.js](https://github.com/tianhaoyao/chat-app/blob/43827468e4f6b646b38b1097c4c0cf56223db2b5/client/src/themes/theme.js) defined for this app, you should try to add reusable color/font/etc. values to the theme and using them in your makeStyles to make the styling more maintainable!

tianhaoyao commented:
![refactor](https://user-images.githubusercontent.com/32031491/117500862-3d40d880-af4b-11eb-8778-111c053f3364.png)
![register](https://user-images.githubusercontent.com/32031491/117500865-3dd96f00-af4b-11eb-9631-f77fd1eb983d.PNG)
![login](https://user-images.githubusercontent.com/32031491/117500866-3dd96f00-af4b-11eb-96e4-fa71d16c08e6.PNG)

Screenshots attached are what the login and signup pages, and how I refactored to avoid duplicate code . I tried to utilize reusable colors/such a bit, not sure if it's enough.

tmattmac commented on client/src/SignupForm.js:
```diff
@@ -0,0 +1,173 @@
+    maxWidth: '200px',
+    maxHeight: '70px',
+    minWidth: '200px',
+    minHeight: '70px',
```
You can just set `width` and `height` instead here.

tmattmac commented on client/src/LoginForm.js:
```diff
@@ -0,0 +1,149 @@
+                      type="password"
+                      name="password"
+                      InputProps={{endAdornment:
+                      <Button>
+                        <Typography variant="h6" color="primary" disableFocusRipple disableRipple>
+                          Forgot?
+                        </Typography>
+                      </Button>}
```
You don't have to include the "Forgot" button, but in the future, you should wrap this in an [`InputAdornment`](https://material-ui.com/components/text-fields/#input-adornments) component.

tmattmac reviewed (commented):
Changes look good, just a few notes:

Make sure you're consistently using classes instead of inline styles.

The buttons are a bit large; the text should be about the size of the text in the input fields.

Also, I noticed that some of the indenting for the JSX isn't lining up. It's recommended to use a code formatter like Prettier if you find it troublesome to keep track of indentation.

I'm also noticing a lot of extra spaces between lines. Adding line breaks within functions/components is fine to distinguish connected blocks of code, but try to avoid using more than one line break where necessary, and avoid doing things like this:

![image](https://user-images.githubusercontent.com/30205106/117515884-8c940280-af65-11eb-801d-32f73f09c6d6.png)

tianhaoyao commented:
![login](https://user-images.githubusercontent.com/32031491/117520903-d508eb80-af78-11eb-9a65-db3f81bd7e46.PNG)
![register](https://user-images.githubusercontent.com/32031491/117520904-d5a18200-af78-11eb-9373-3640ec43d81c.PNG)

I have shrunk the button text sizes to match the input text size, removed the forgot password button, moved inline styles to classes, and formatted everything with Prettier. Used width and height instead of max/min width/height.
